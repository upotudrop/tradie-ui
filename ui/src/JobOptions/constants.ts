export const INVITED_JOB_OPTION = 'Invited';
export const ACCEPTED_JOB_OPTION = 'Accepted';