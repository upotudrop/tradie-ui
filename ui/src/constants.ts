export const JOB_OPTION_INVITED_KEY = 'Invited';
export const JOB_OPTION_ACCEPTED_KEY = 'Accepted';

export const ACCEPTED_JOB_STATUS = 'accepted';
export const DECLINED_JOB_STATUS = 'declined';