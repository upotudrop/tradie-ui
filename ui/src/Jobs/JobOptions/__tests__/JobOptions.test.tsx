import { JobOptions } from '../index';
import { render, act } from '@testing-library/react'
import { ACCEPTED_JOB_OPTION, INVITED_JOB_OPTION } from '../constants';

describe('Job Options Testing suite', () => {

    it('should be able to render tradie ui', async () => {
        const { container, getByText } = await act(async () => render(
        <JobOptions 
            children={<></>} 
            onChangeOption={jest.fn()}
            optionKey={INVITED_JOB_OPTION} />
            )
        )

        expect(getByText(INVITED_JOB_OPTION)).toBeInTheDocument();
        expect(getByText(ACCEPTED_JOB_OPTION)).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    })
 
})
