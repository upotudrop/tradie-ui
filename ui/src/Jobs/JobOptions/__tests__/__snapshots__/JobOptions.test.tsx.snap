// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Options Testing suite should be able to render tradie ui 1`] = `
<div>
  <section
    class="job-options-container"
  >
    <section
      class="selected-option-tab"
    >
      <p>
        Invited
      </p>
    </section>
    <section
      class="default-option-tab"
    >
      <p>
        Accepted
      </p>
    </section>
  </section>
</div>
`;
